timemachine/fe/free_energy.py

```diff
@@ -452,11 +455,24 @@ def prepare_host_edge(self, ff: Forcefield, host_config: HostConfig, lamb: float
         host_bps, host_masses = openmm_deserializer.deserialize_system(host_config.omm_system, cutoff=1.2)
         hgt = topology.HostGuestTopology(host_bps, self.top, host_config.num_water_atoms, ff, host_config.omm_topology)
 
-        final_params, final_potentials = self._get_system_params_and_potentials(ff_params, hgt, lamb)
-        combined_masses = self._combine(ligand_masses, host_masses)
-        return final_potentials, final_params, combined_masses
+        final_params = []
+        final_potentials = []
+        combined_params, combined_potentials = self._get_system_params_and_potentials(ff_params, hgt, lamb)
+        for params, pot in zip(combined_params, combined_potentials):
+            # Unpack the summed potential to be consistent with SingleTopology and so
+            # that downstream code which relies on the potential types works properly
+            # TBD: Deboggle and unify the topology classes
+            if isinstance(pot, SummedPotential):
+                for partial_params, sub_pot in zip(pot.params_init, pot.potentials):
+                    final_params.append(partial_params)
+                    final_potentials.append(sub_pot)
```

Comment:
This is correct for one level of nesting -- perhaps raise error if `isinstance(sub_pot, SummedPotential)` ?

Reply:
Added in https://github.com/proteneer/timemachine/pull/1431/commits/9d67be72ea9641cbabe4246c9e59d4c9ad8a5914https://github.com/proteneer/timemachine/pull/1431/commits/9d67be72ea9641cbabe4246c9e59d4c9ad8a5914